$MirOS: ports/print/cups/patches/patch-backend_serial_c,v 1.3 2006/11/08 18:25:44 bsiegert Exp $
--- backend/serial.c.orig	Sat Jul 12 00:48:26 2008
+++ backend/serial.c	Tue Oct 21 22:56:22 2008
@@ -749,7 +749,7 @@ list_devices(void)
 
   for (i = 0; i < 100; i ++)
   {
-    sprintf(device, "/dev/ttyS%d", i);
+    snprintf(device, sizeof(device), "/dev/ttyS%d", i);
 
     if ((fd = open(device, O_WRONLY | O_NOCTTY | O_NDELAY)) >= 0)
     {
@@ -786,9 +786,10 @@ list_devices(void)
     }
   }
 
+# if (defined(__OpenBSD__))||(defined(__MirBSD__))||(defined(__ekkoBSD__))
   for (i = 0; i < 16; i ++)
   {
-    sprintf(device, "/dev/usb/ttyUSB%d", i);
+    snprintf(device, "/dev/ttyU%d", i);
     if ((fd = open(device, O_WRONLY | O_NOCTTY | O_NDELAY)) >= 0)
     {
       close(fd);
@@ -819,6 +820,18 @@ list_devices(void)
       }
     }
   }
+# else
+  for (i = 0; i < 16; i ++)
+  {
+    sprintf(device, "/dev/usb/ttyUSB%d", i);
+    if ((fd = open(device, O_WRONLY | O_NOCTTY | O_NDELAY)) >= 0)
+    {
+      close(fd);
+      printf("serial serial:%s?baud=230400 \"Unknown\" \"USB Serial Port #%d\"\n",
+             device, i + 1);
+    }
+# endif
+
 #elif defined(__sgi)
   int		i, j, n;	/* Looping vars */
   char		device[255];	/* Device filename */
@@ -1044,7 +1057,7 @@ list_devices(void)
 
   for (i = 0; i < 32; i ++)
   {
-    sprintf(device, "/dev/ttyd%c", funky_hex[i]);
+    snprintf(device, sizeof(device), "/dev/ttyd%c", funky_hex[i]);
     if ((fd = open(device, O_WRONLY | O_NOCTTY | O_NDELAY)) >= 0)
     {
       close(fd);
@@ -1060,7 +1073,7 @@ list_devices(void)
   for (i = 0; i < 16; i ++) /* Should be up to 65536 boards... */
     for (j = 0; j < 32; j ++)
     {
-      sprintf(device, "/dev/ttyc%d%c", i, funky_hex[j]);
+      snprintf(device, sizeof(device), "/dev/ttyc%d%c", i, funky_hex[j]);
       if ((fd = open(device, O_WRONLY | O_NOCTTY | O_NDELAY)) >= 0)
       {
 	close(fd);
@@ -1084,7 +1097,7 @@ list_devices(void)
   for (i = 0; i < 16; i ++) /* Should be up to 65536 boards... */
     for (j = 0; j < 32; j ++)
     {
-      sprintf(device, "/dev/ttyD%d%c", i, funky_hex[j]);
+      snprintf(device, sizeof(device), "/dev/ttyD%d%c", i, funky_hex[j]);
       if ((fd = open(device, O_WRONLY | O_NOCTTY | O_NDELAY)) >= 0)
       {
 	close(fd);
@@ -1099,7 +1112,7 @@ list_devices(void)
 
   for (i = 0; i < 32; i ++)
   {
-    sprintf(device, "/dev/ttyE%c", funky_hex[i]);
+    snprintf(device, sizeof(device),"/dev/ttyE%c", funky_hex[i]);
     if ((fd = open(device, O_WRONLY | O_NOCTTY | O_NDELAY)) >= 0)
     {
       close(fd);
@@ -1114,7 +1127,7 @@ list_devices(void)
 
   for (i = 0; i < 128; i ++)
   {
-    sprintf(device, "/dev/ttyA%d", i + 1);
+    snprintf(device, sizeof(device), "/dev/ttyA%d", i + 1);
     if ((fd = open(device, O_WRONLY | O_NOCTTY | O_NDELAY)) >= 0)
     {
       close(fd);
